### PR TITLE
Related blueprints

### PR DESCRIPTION
**Pull request by bramkragten**
## Proposed change

Needs: https://github.com/home-assistant/core/pull/78535

- Allows to see all the automations/scripts that are using a certain blueprint
- Add the used blueprint to related items of the script/automation
- Lists the automations/scripts that prevent a blueprint from being deleted

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/blueprint/ha-blueprint-overview.ts [hunk 1]
```diff
@@ -326,7 +327,19 @@ class HaBlueprintOverview extends LitElement {
...
   private _delete = async (ev) => {
-    const blueprint = ev.currentTarget.blueprint;
+    const blueprint = ev.currentTarget.blueprint as BlueprintMetaDataPath;
+    const related = await findRelated(
+      this.hass,
+      `${blueprint.domain}_blueprint`,
+      blueprint.path
+    );
+    if (related.automation?.length || related.script?.length) {
+      showAlertDialog(this, {
```
Comment (bramkragten) [inline src/panels/config/blueprint/ha-blueprint-overview.ts:337]: Should we ask the user if we should delete all scripts/automations that use the blueprint for them? Seems a bit scary...

File: src/panels/config/blueprint/ha-blueprint-overview.ts [hunk 2]
```diff
@@ -326,7 +327,19 @@ class HaBlueprintOverview extends LitElement {
...
+        title: "This blueprint is in use, and can not be deleted",
+        text: `Please remove all automations and scripts that use this blueprint before deleting it. Used in: ${related.automation}${related.script}`,
```
Comment (bramkragten) [inline src/panels/config/blueprint/ha-blueprint-overview.ts:339]: TODO: formatting

**Comment by bramkragten** [comment]
In a follow up PR I'll convert the blueprints datatable to use an overflow like the automations/scripts one

File: src/panels/config/blueprint/ha-blueprint-overview.ts (hunk 1 above)
Comment (bramkragten) [reply src/panels/config/blueprint/ha-blueprint-overview.ts]: Convert? What do you mean with that?

**Review by piitaya — APPROVED** [review]

**Pull request merged**